hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/BucketAssignFunction.java

```diff
@@ -141,27 +142,41 @@ public void snapshotState(FunctionSnapshotContext context) {
 
   @Override
   public void initializeState(FunctionInitializationContext context) {
-    ValueStateDescriptor<HoodieRecordGlobalLocation> indexStateDesc =
+    ValueStateDescriptor<HoodieRecord> indexStateDesc =
```

Comment:
This would increase the state size significantly. We better avoid this with better solution, and why we must sore full record instead of index here ?